### ISSUE TITLE
Distillation Tower output enhancement

### ISSUE THREAD

**Issue by AleksSher** [Flaxbeard/ImmersivePetroleum#60]
As of 1.1.0 update distillation tower produces 3 types of liquids, but has only one fluid output valve.
Maybe there is a point to make independent output valves for each distillation product to negate problems with multiple liquids in single pipe system between tower and fluid storage?

**Comment by calloatti** [comment]
And what happens when more outputs are added in the future or thru config changes? Just use an IE fluid router on the output.

**Comment by Flaxbeard** [comment]
I'm inclined to agree with @calloatti here - though one output port is a bit annoying, a fluid router should be all that you need

**Comment by kog01** [comment]
Adding a pump to the pipes containing the three liquids (before the router) will break the flow. If you add a barrel between the pump and the liquid router exactly 1mb of whatever liquid fill flow into it, resuming the flow until it will get clogged again with 1mb of liquid.

**Comment by Frontrider** [comment]
The real problem, is that its unintuitive. If it was, this report would't be here.
If there was an intuitive method, there would be no problem with the single fluid output.

Maybe,do take inspiration from an another mod, which has a block storing mutiple liquids "in the same tank":
Tinker's Construct.
Only output the bottom liquid. It makes sure, that only a single liquid is visible to any pipes, and also makes it easy to predict whitch liquid is gonna go out.
Add a little delay between operations (don't produce constantly, take breaks), so we can empty the tank whitout a risk of getting stuck on constatly extracting 1mbs of fluid. 
At this point, if his pipes get clogged up (because he's not routing it properly), than thats not your problem anymore.

**Comment by Flaxbeard** [comment]
I'll play around this more since people feel strongly about it - what transport methods would you like to work that currently don't?

**Comment by Frontrider** [comment]
If it works with thermal dynamics, than it should work with all the things.

**Comment by Eufranio** [comment]
I'd rather prefer a new output to the tower. This is a Immersive Engineering addon, not thermal dynamics one, it is supposed to be playable with ONLY IE and the mod. Atm, I cannot use the distillation tower at 100% because of the fluid pipes problem: they won't extract 100% of the liquid because there are a lot of liquids going around.

**Comment by FnordMan** [comment]
Yeah, same thoughts here the distillation tower is basically unusable due to how it outputs. Most of the time it won't output things in the ratio the book says it should. 

It would be better (and more realistic) to have three separate outputs, one for each fluid type.

**Comment by benpankow** [comment]
If you're playing with the new version I'm curious as to what your problems are - I reworked the system and it seems to work fine using IE pipes and TE pipes

**Comment by Flaxbeard** [comment]
If you're using the newest version, I'm curious as to what the problems are. I've tested the updated output with TE and IE pipes w/ fluid router and both work fine.

Multiple outputs would be a solution that would limit customization - what if packs add different recipes

**Comment by Eufranio** [comment]
I'm playing on 1.12, with the latest version. By what it seems, only two of the three fluids get sent quick enough, and one of them stays inside the tower tank. On my case, even with a fluid router and 3 different pipe lines, gasoline always stayed inside the tower.

**Comment by Eufranio** [comment]
The tower could have one product output and one byproduct output. Byproduct outputs would be able to dispense more than one fluid, while the product one would only dispense the main result (on this case, diesel).

**Comment by mgwerner** [comment]
1.1.9.  Only one fluid is being pumped out of the tower. the other two fill up but do not pump out.

I have tried fluid router at the output, and along the pipeline, broken and reset the multiblock.  Broken and relaid the pipe, with and without a pump.  Only using IE for power, routing and piping.  Bitumen is being produced.

Frustrating.  I recall several iterations ago this worked well.

**Comment by Tech0verlord** [comment]
Fluid is pretty much stuck in the tower. I've tried placing a fluid router directly in front of the tower's output, but, besides routing fluids, it doesn't help the fact that either the router or the tower itself gets jammed and won't output anything. I've tried IE's fluid pipes, various fluiducts from Thermal Expansion (which have no minimum for fluid movement IIRC), and nothing. We either need a way to dump excess fluids, more outputs, or an internal buffer somewhere to prevent this from happening.

**Comment by bjhemsworth** [comment]
A helpful comment for Googlers who end up here while searching for a solution to the 1.1.9 distilation tower output glitch.

**Using a seperate fluid router for each liquid clears up the problem, have the mixed liquid leave the tower via a regular fluid pipe then attach 3 fluid routers to it to sort fluid seperatly prevents them becoming jammed.**

Related to this thread and the discussion of why seperate outputs are needed. Its not just thematically appropriate but the tower is actually outputting the liquids one at a time through the same pipe. The current system is therefore delivering the materials in unpredicatable batches through the pipe one at a time, which could be a problem for anyone creating a highly balanced interlocking build with them that realies on a consistent flow of one resource. For me personally it just makes it really difficult to gauge how much of each resource I'm consuming and producing and if I have a build up somewhere that needs more storage or adjustment

**Comment by Sabrewulf** [comment]
It really need 1 output for each byproduct instead create more complexity.
It's like to connect sewage to potable water, makes no sense and is problematic.

**Comment by voidsong-dragonfly** [comment]
Agreeing with Flaxbeard and Calloatti, this system should be kept how it is. Currently, it allows for any number of outputs as long as the total throughput doesn't exceed what it can output per tick. Having a set number of outputs (unless the number is very large) makes adding recipes with more than three fluid outputs an incredibly excruciating process, as one suddenly has to separate the outputs into multiple intermediate fluids and build multiple towers to get the full spread of products. I routinely have recipes with >6 fluids and sometimes 9, so the current system is perfect for this.

**Comment by bjhemsworth** [comment]
An enhancement suggestion.

A modular output block that attaches, in some capacity, to the distilation tower, providing an output port for a single product each. That way any number of physical output ports could be added by the user. Think of it as a more thematically appropriate fluid router.
(The enduring popularity of this thread is partly down to the repeated bugs and compatability problems with fluid routers from existing mods, and the non intuative way the tower outputs its products down a single pipe)

Being able to place our own outputs could even possibly mean being able to put the outputs on the appropriate stage of the fractioning column (lpg output at the top,  bitumin output at the bottom, etc)

**Comment by xienwolf** [comment]
I am also here due to the absolute nuisance of pulling outputs from this tower.  I am now trying to figure out how the fluid router works because my output is NEARLY full with lubricant, but I can only pull out the blue stuff.  Soon there won't be a full bucket worth of space for the blue to be pulled out.

Things that work in other mods:
Smeltery option of clicking a fluid to make it move to the "front" of the exit line already has been mentioned.
One mod has an eye dropper you can click on a bar to pull whichever fluid you click on.  Maybe make something like that possible with standard buckets?
If there is an empty bucket, attempt to output whatever is "in front" for liquids, but if there isn't a full bucket of that, move on to the next liquid, then the next after that.
A button to void all of the liquid present at the front of the line.  Then you can load an empty bucket, click the button, and get a bucket of the liquid in second position.

**Comment by mrh0** [comment]
Yeah, I'm having a lot of problems with it just not outputting, even with a fluid router, It will work for a bit but then suddenlys stop outputting.

**Comment by Solverbolt** [comment]
I am new here, but I am running into a problem. I gave it 3 outputs, as the quest line states for ATM 10, but its only producing 1 fluid. 

 With Shale Oil, it only produces helium, nothing else.
Crude Oil = Sulfuric Naptha
 
I had the same problem when I tried to process in the more base level distillery.  It just ignores all other outputs as valid
